### PR TITLE
export `offline-worker` as HTTP API

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 

export `offline-worker` as HTTP API for testing.

### What is changed and how it works?

add gRPC gateway option in proto.

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 bin
 */*.iml
+.DS_Store
```
Comment (csuzhangxc) [inline .gitignore:3]: 🤔

Comment (lichunzhu) [reply .gitignore:3]: It's a hidden file generated by mac os which bothers me for a long time. 😕I can delete this as well.

Comment (csuzhangxc) [reply .gitignore:3]: > which bothers me for a long time

😢

**Pull request merged**